docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+> important
+
+Example:
+
+> We use this method because Chrome's built-in PDF viewer preserves hyperlinks.
+
+<br>
+
+> :exclamation: Text that looks like this, beginning with the :exclamation: sign indicates information that is very
+> important, such as warnings about potential mistakes or common problems
+
+Example:
+
+> :exclamation: **Caution** Follow the steps in the following guide precisely.
+
+## 2. Setting up
+
+The following section describes how to set up the coding environment on your own computer, in order to start writing
+code to improve Zer0Note.
+
+### 2.1. Prerequisites
+
+1. **JDK** 11
+2. **IntelliJ** IDEA
+
+### 2.2. Setting up the project in your computer
+
+>:exclamation: **Caution:** Follow the steps in the following guide precisely.
+>Things will not work out if you deviate in some steps.
+
+1. **Fork** this repo, and **clone** the fork into your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   a. Click `Configure` > `Project Defaults` > `Project Structure`  
+   b. Click `New...` and find the directory of the JDK.
+4. Click `Import Project`
+5. Locate the `build.gradle` file and select it. Click `OK`.
+6. Click `Open as Project`.
+7. Click `OK` to accept the default settings.
+
+### 2.3. Verifying the setup
+
+1. Run the `seedu.duke.Duke`.
+2. Try a few commands.
+3. [Run the tests](#61-running-tests) to ensure they all pass.
+
+### 2.4. Configure the coding style
+
+If using IDEA, follow the guide [[se-edu/guides] IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html)
+to set up IDEA’s coding style to match ours.
+
+>Optionally, you can follow the guide [[se-edu/guides] Using Checkstyle](https://se-education.org/guides/tutorials/checkstyle.html)
+>to find how to use the CheckStyle within IDEA e.g., to report problems as you write code.
+
+## 3. Design
+
+The following section describes the design and implementation of the product. UML diagrams and code snippets are used
+to explain some aspects of the code. If you are unfamiliar with UML, the diagrams should still be fairly
+understandable. However, you may wish to consult [[CS2113/T] Modeling](https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/modeling.html) for a quick introduction to UML.
+
+### 3.1 Architecture (Neil)
+
+**How the architecture components interact with each other**
+The Sequence Diagram below shows how the components interact with each other for the scenario...
+
+/* work in progress */
+
+### 3.2 UserInterface Component (Neil)
+
+The UserInterface Component is made up of `AppMode`, `AppState`, `CliMessages`, `CliUserInterface`, `InputParser`.
+
+The `UserInterface` component,
+
+* Executes user commands using the `Commands` component.
+* // how it interacts with the other components
+
+/* TODO explain the various variables and methods */
```

Comment:
Will this section be explaining the UML diagram in section 3.3? If not then I think it would be beneficial to have a brief description of how the different components of Command (and other like Tasks, Notebooks, etc) work together.

Reply:
3.3 will indeed explain the UML diagrams in that section, yes. We didn't realise it needed to be done for today so bits of it like this one are incomplete. Sorry about that.